lib/web/fetch/util.js

```diff
@@ -537,45 +572,133 @@ function stripURLForReferrer (url, originOnly) {
   return url
 }
 
-function isURLPotentiallyTrustworthy (url) {
-  if (!webidl.is.URL(url)) {
+const potentialleTrustworthyIPv4RegExp = new RegExp('^(?:' +
+  '(?:127\\.)' +
+  '(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\\.){2}' +
+  '(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[1-9])' +
+')$')
+
+const potentialleTrustworthyIPv6RegExp = new RegExp('^(?:' +
+  '(?:(?:0{1,4}):){7}(?:(?:0{0,3}1))|' +
+  '(?:(?:0{1,4}):){1,6}(?::(?:0{0,3}1))|' +
+  '(?:::(?:0{0,3}1))|' +
+')$')
```

Comment:
Because isIPv4 and isIPv6 are just validating that they are any IPv4 and any IPv6. The spec explicitly says IPs matching CIDR 127.0.0.1/8 and ::1/128. 

127.0.0.1/8 is 127.0.0.1 - 127.0.0.255
::1/128 is just ::1 and all its variations, like 0:0:0:0:0:0:0:1, 0::1 etc..

The previous regex in the code was wrong.